husarion_ugv_utils/include/husarion_ugv_utils/moving_average.hpp

```diff
@@ -25,40 +25,40 @@ class MovingAverage
 {
 public:
   MovingAverage(const std::size_t window_size = 5, const T initial_value = T(0))
-  : window_size_(window_size), initial_value_(initial_value), sum_(T(0))
+  : window_size_(window_size), initial_value_(initial_value)
   {
   }
 
   void Roll(const T value)
   {
     values_.push_back(value);
-    sum_ += value;
 
     if (values_.size() > window_size_) {
-      sum_ -= values_.front();
       values_.pop_front();
     }
   }
 
-  void Reset()
-  {
-    values_.erase(values_.begin(), values_.end());
-    sum_ = T(0);
-  }
+  void Reset() { values_.erase(values_.begin(), values_.end()); }
 
   T GetAverage() const
   {
     if (values_.size() == 0) {
       return initial_value_;
     }
-    return sum_ / static_cast<T>(values_.size());
+
+    T sum = T(0);
+
+    for (const auto & value : values_) {
+      sum += value / static_cast<T>(values_.size());
+    }
+
+    return sum;
```

Comment:
Devision is computation expensive we can first acumulate buffor and then devide. Reduce devision from N -> 1.
```suggestion
    T sum = std::accumulate(buffer_.begin(), buffer_.end(), 0);
    T average = sum / static_cast<T>(values_.size());

    return average;
```

Reply:
I did it this way to improve numeric stability for floats with large accumulate buffer values, but honestly without using recursive average or more numerically stable summation methods it does not matter that much.